apis/field_error.go

```diff
@@ -55,17 +55,16 @@ func (fe *FieldError) ViaField(prefix ...string) *FieldError {
 		return nil
 	}
 	newErr := &FieldError{}
-	for _, e := range fe.getNormalizedErrors() {
-		// Prepend the Prefix to existing errors.
-		newPaths := make([]string, 0, len(e.Paths))
-		for _, oldPath := range e.Paths {
-			newPaths = append(newPaths, flatten(append(prefix, oldPath)))
-		}
-		sort.Slice(newPaths, func(i, j int) bool { return newPaths[i] < newPaths[j] })
-		e.Paths = newPaths
-
-		// Append the mutated error to the errors list.
-		newErr = newErr.Also(&e)
+	// Prepend the Prefix to existing errors.
+	newPaths := make([]string, 0, len(fe.Paths))
+	for _, oldPath := range fe.Paths {
+		newPaths = append(newPaths, flatten(append(prefix, oldPath)))
+	}
+	newErr.Details = fe.Details
```

Comment:
nit: could be extracted in a function : 

```go
func From(fe *FieldError, ops ...func(*FieldError)) *FieldError {
    newErr := &FieldError{
        Details: fe.Details,
        Message: fe.Message,
        Paths: fe.Paths,
    }
    for _, op := range ops {
        op(fe)
    }
    return newErr
}
// […]
newErr := From(fe, func(e *FieldError) { e.Paths = newPaths })
// this func could even be extracted in a `withPaths` func
```

Reply:
I am not sure I follow the example above. I have updated the code to use deepcopy on .Also, and simplified the construction of the newFe inside of .ViaField. Maybe this is in-line with what you are suggesting

Reply:
@n3wscott yeah even better, forgot about those generation and deepcopy stuff :wink: 
:+1: